src/pages/SwapNew/components/SwapInfo/SwapInfoBasics.tsx

```diff
@@ -0,0 +1,93 @@
+
+import { ReactComponent as DownArrowSmallSVG } from '../../../../assets/swapbox/down-arrow-small.svg'
+import { useGasInfo } from '../../../../hooks/useGasInfo'
+import { useUserSlippageTolerance } from '../../../../state/user/hooks'
+import { IndicatorColorVariant, IndicatorIconVariant } from '../../constants'
+import { CurrenciesConversionRate } from './CurrenciesConversionRate'
+import { Indicator } from './Indicator'
+
+type SwapInfoBasicsProps = {
+  selectedTrade?: Trade
+  allPlatformTrades?: (Trade | undefined)[]
+  showSwapInfoDetails: boolean
+  toggleShowInfoDetails: () => void
+}
+
+export function SwapInfoBasics({
+  selectedTrade,
+  allPlatformTrades,
+  showSwapInfoDetails,
+  toggleShowInfoDetails,
+}: SwapInfoBasicsProps) {
+  const getPlatformsColor = () =>
+    allPlatformTrades!.filter(Boolean).length / allPlatformTrades!.length > 0.5
+      ? IndicatorColorVariant.POSITIVE
+      : IndicatorColorVariant.NEGATIVE
+  const getNumberOfPlatforms = () => `${allPlatformTrades!.filter(Boolean).length}/${allPlatformTrades!.length}`
+
+  const userSlippageTolerance = useUserSlippageTolerance()
+  const { gas } = useGasInfo()
+
+  return (
+    <Container>
+      <SwapCostInfo>
+        {allPlatformTrades?.length !== 0 && (
+          <Indicator
+            color={getPlatformsColor()}
+            icon={IndicatorIconVariant.MAGNIFYING_GLASS}
+            text={getNumberOfPlatforms()}
+          />
+        )}
+        <Indicator
+          color={IndicatorColorVariant.WARNING}
+          icon={IndicatorIconVariant.GAS}
+          text={gas?.normal?.toString()}
+        />
+        <Indicator
+          color={!!userSlippageTolerance ? IndicatorColorVariant.POSITIVE : IndicatorColorVariant.NEGATIVE}
+          icon={IndicatorIconVariant.BANANA}
+          text={`${(userSlippageTolerance / 100).toFixed(1)}%`}
+        />
+        <Indicator
+          color={selectedTrade instanceof CoWTrade ? IndicatorColorVariant.POSITIVE : IndicatorColorVariant.UNDEFINED}
+          icon={IndicatorIconVariant.SHIELD}
+        />
+        <CurrenciesConversionRate price={selectedTrade?.executionPrice} />
+      </SwapCostInfo>
+      <ExpandButton onClick={toggleShowInfoDetails}>
+        <DownArrowSmall showSwapInfoDetails={showSwapInfoDetails} />
+      </ExpandButton>
+    </Container>
+  )
+}
+
+const Container = styled.div`
+  width: 100%;
+  height: 20px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`
+
+const SwapCostInfo = styled.div``
+
+const ExpandButton = styled.button`
+  height: 20px;
+  width: 20px;
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  padding: 0;
+  background: transparent;
+  border-radius: 4px;
+  border: none;
+  cursor: pointer;
+`
+
+const DownArrowSmall = styled(DownArrowSmallSVG)<{ showSwapInfoDetails: boolean }>`
+  transition: all 0.2s ease;
+  transform: ${({ showSwapInfoDetails }) => (showSwapInfoDetails ? 'rotate(180deg)' : 'rotate(0deg)')};
+`
```

Comment:
```suggestion
const DownArrowSmall = styled(DownArrowSmallSVG).withConfig({
  shouldForwardProp: (prop, fn) => fn(prop),
})<{ showSwapInfoDetails: boolean }>`
transition: all 0.2s ease;
  transform: ${({ $showSwapInfoDetails }) => ($showSwapInfoDetails ? 'rotate(180deg)' : 'rotate(0deg)')};
`
```

or add `$ ` so it becomes  Transient props [link](https://styled-components.com/docs/api#transient-props)

```suggestion
const DownArrowSmall = styled(DownArrowSmallSVG)<{ $showSwapInfoDetails: boolean }>`
  transition: all 0.2s ease;
  transform: ${({ $showSwapInfoDetails }) => ($showSwapInfoDetails ? 'rotate(180deg)' : 'rotate(0deg)')};
`
```


This will prevent props from going down to component

Reply:
Yeah, I agree. 👍🏼

Reply:
Done.